src/test/shared/vscode/runCommand.test.ts

```diff
@@ -91,5 +91,102 @@ describe('runCommand', function () {
+            })
+
+            it('throttles events when argIndexes not set', async function () {
+                const command = Commands.register('_test.telemetryName.noUniqueIndexes', (obj: any) => {})
+
+                await command.execute({ id: 'a' })
+                // These will be throttled
+                await command.execute(undefined)
+                await command.execute('a')
+                await command.execute('b')
+
+                const events = globals.telemetry.logger.query({ metricName: 'vscode_executeCommand' })
+                assert.strictEqual(events.length, 1)
+            })
+
+            it('correctly increments the debounce counter when argIndexes is set', async function () {
+                const command = Commands.register(
+                    {
+                        id: '_test.telemetryName.commandSourceArgDebounceCount',
+                        argIndexes: {0: 'source'},
+                    },
+                    (source: string) => {}
+                )
+
+                const clock = installFakeClock()
+
+                await command.execute('a')
+                // following 3 events will be throttled
+                await command.execute('a')
+                await command.execute('a')
+                await command.execute('a')
+                // advance clock past throttle time
+                clock.tick(defaultTelemetryThrottleMs)
+                // will contain a 'debounceCount' of 3
+                await command.execute('a')
+
+                const events = globals.telemetry.logger.query({ metricName: 'vscode_executeCommand' })
+                assert.strictEqual(events.length, 2)
+
+                const firstEvent = events[0]
+                assert.strictEqual(firstEvent['source'], 'a')
+                assert.strictEqual(firstEvent['debounceCount'], undefined)
+
+                const secondEvent = events[1]
+                assert.strictEqual(secondEvent['source'], 'a')
+                assert.strictEqual(secondEvent['debounceCount'], '3')
+            })
+
+            it('throws if out of index', async function () {
+                const command = Commands.register(
+                    {
+                        id: '_test.telemetryName.throwsOnOutOfIndex',
+                        argIndexes: {1: 'source'}, // out of index
+                    },
+                    (source: any) => {}
+                )
+                await assert.rejects(() => command.execute(1), new ToolkitError(`Unique arg indexes exceed the # of args for: "_test.telemetryName.throwsOnOutOfIndex"`))
+            })
+
+            it('throws if arg type is not supported', async function () {
+                const command = Commands.register(
+                    {
+                        id: '_test.telemetryName.throwsOnUnsupportedType',
+                        argIndexes: {0: undefined},
```

Comment:
we want to communicate that this defines a composite key. so let's use a name that makes that clear. there is precedent for simply calling it `key` in e.g. [python](https://docs.python.org/3/howto/sorting.html#key-functions). or we can call it `compositeKey`.
  
```suggestion
                        compositeKey: {0: undefined},
```